obspy/io/rg16/core.py

```diff
@@ -1,233 +1,845 @@
-    chan = BAND_MAP[sampling_rate] + INSTRUMENT_CODE + component
-
-    statsdict = dict(
-        starttime=UTCDateTime(theader['time'] / 1000000.),
-        sampling_rate=sampling_rate,
-        npts=theader['samples'],
-        network=str(theader['line_number']),
-        station=str(theader['point']),
-        location=str(theader['index']),
-        channel=chan,
-    )
-    return Stats(statsdict)
 
+def _read_extended_headers(fi):
+    """
+    Extract informations from the extended headers.
+    """
+    extended_headers = {}
+    nbr_channel_set = _read(fi, 28, 1, 'bcd')
+    start_byte = 32 + 32 + 32*nbr_channel_set
+    extended_headers['1'] = _read_extended_header_1(fi, start_byte)
+    start_byte += 32
+    extended_headers['2'] = _read_extended_header_2(fi, start_byte)
+    start_byte += 32
+    extended_headers['3'] = _read_extended_header_3(fi, start_byte)
+    nbr_extended_headers = _read(fi, 37, 2, 'binary', True)
+    if nbr_extended_headers > 3:
+        coeffs = extended_headers['2']['number_decimation_filter_coefficient']
+        nbr_coeff_remain = coeffs % 8
+        for i in range(3, nbr_extended_headers):
+            start_byte += 32
+            extended_header_name = str(i+1)
+            if i == nbr_extended_headers - 1:
+                header = _read_extended_header(fi, start_byte, i+1,
+                                               nbr_coeff_remain)
+                extended_headers[extended_header_name] = header
+            else:
+                header = _read_extended_header(fi, start_byte, i+1, 8)
+                extended_headers[extended_header_name] = header
+    return extended_headers
 
-# Note: I am leaving this function in the code as it may be needed if the
-# "read until end of file" approach needs to be revised. Also, there are
-# some non-obvious peculiarities about the rg16 files I have worked with
-# that are handled by this function.
-
-# def _get_num_traces(fi, byte_start, gheader, eheader):
-#     """
-#     Get the number of traces contained in this file by reading trace sets.
-#
-#     Note: This function was created because multiplying channel_sets in
-#     the general header by num_records in the extended header doesn't work for
-#     some larger files.
-#     """
-#     channel_sets = gheader['channel_sets']
-#     num_records = eheader['num_records']
-#
-#
-#     # try reading the channel_header blocks. This is seems to be correct
-#     # when there are millions of records in the file
-#     channel_dicts = [read_block(fi, channel_header_block, byte_start + x*32)
-#                      for x in range(channel_sets)]
-#     num_traces1 = np.sum([x['num_channels'] for x in channel_dicts])
-#
-#     # try multiplying general_header and num_records. This seems to be right
-#     # when there arent that many treaces in the file
-#     num_traces2 = channel_sets * num_records
-#
-#     return max(num_traces1, num_traces2)
-
-
-if __name__ == '__main__':
```

Comment:
These lines should be at the end of each file. They run the doctests (when applicable)